woocommerce/woocommerce
# Update `wp-env` To 7.0.0

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This pull request updates `wp-env` to version 7.0.0. This new version includes a couple of upstream changes that we can utilize to simplify the bootstrapping of our E2E tests:

* https://github.com/WordPress/gutenberg/pull/50196: This will allow us to automatically run the E2E setup script without having to do so explicitly.
* https://github.com/WordPress/gutenberg/pull/49962: This fixes the permission errors that required us to manually set some of them to get things working.
* https://github.com/WordPress/gutenberg/pull/41179: This removes the need for quoting `run` arguments. There seems to be a bug with quotes being removed unless escaped, but, this is a nice little touch.

I've gone ahead and made some changes to the E2E bootstrapping to utilize these. We also _technically_ don't need the `update-wp-env.php` script at all, but, I wanted to keep that in place for this pull request. A subsequent one can remove that once the update is in place.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure E2E tests pass in CI.
2. Run `pnpm --filter=woocommerce env:dev` to update the environment with the changes that were made.
3. Test out the environment and make sure it still works as expected.
